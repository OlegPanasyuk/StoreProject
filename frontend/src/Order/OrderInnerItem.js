import React, { Component } from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';


export class OrderInnerItem extends Component {
    render() {
        let { obj } = this.props;
        return (

            <Card className={'p-0 '}>
                <Card.Header>
                    <Card.Text>
                        {obj.name}
                    </Card.Text>
                </Card.Header>
                <Card.Body>
                    <Card.Text>
                        Price was:{obj.price}
                    </Card.Text>
                </Card.Body>
            </Card>

        );
    }
}

OrderInnerItem.propTypes = {
    obj: PropTypes.object
};

export default OrderInnerItem;