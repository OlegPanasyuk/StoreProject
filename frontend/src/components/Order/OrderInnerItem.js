import React from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';

export function OrderInnerItem({
    obj
}) {
    return (
        <Card className='p-0'>
            <Card.Header>
                <Card.Text>
                    {obj.name}
                </Card.Text>
            </Card.Header>
            <Card.Body>
                <Card.Text>
                    Price was:
                    {obj.price}
                </Card.Text>
            </Card.Body>
        </Card>
    );
}


OrderInnerItem.propTypes = {
    obj: PropTypes.object
};

OrderInnerItem.defaultProps = {
    obj: {}
};

export default OrderInnerItem;
