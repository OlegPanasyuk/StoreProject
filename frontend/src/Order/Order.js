import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, Collapse, Button, CardGroup } from 'react-bootstrap';
import OrderInnerItem from './OrderInnerItem';

export class Order extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
    }

    toggleCollapse() {
        this.setState((state) => ({
            open: !state.open
        }));
    }

    render() {
        let { obj } = this.props;
        let contents = JSON.parse(obj.contents);
        // let arrControls = contents.map((el) => {

        // });
        // console.log('From Oreder ,with Love',contents);
        return (
            <Card className='mb-3'>
                <Card.Header className='d-flex align-items-center justify-content-between'>
                    <Card.Text className='mb-0'>
                        Order #{obj.id}
                    </Card.Text>
                    <Button
                        variant='light'
                        onClick={() => this.toggleCollapse()}

                        aria-controls={`collapse-cards-${obj.id}`}
                        aria-expanded={this.state.open}
                    >
                        {
                            this.state.open ?
                                <i className="fas fa-chevron-up"></i>
                                :
                                <i className="fas fa-chevron-down"></i>
                        }
                    </Button>
                </Card.Header>

                <Card.Body className={'d-flex flex-wrap justify-content-start' + (this.state.open ? '' : ' p-0')}>
                    <Collapse in={this.state.open}>
                        <CardGroup >
                            {contents.map((el) => {
                                return (

                                    <OrderInnerItem

                                        key={`${el.idgoods}-${el.name}`}
                                        obj={el} />
                                );
                            })}
                        </CardGroup>
                    </Collapse>
                </Card.Body>

            </Card>

        );
    }
}

Order.propTypes = {
    obj: PropTypes.object
};

export default Order;