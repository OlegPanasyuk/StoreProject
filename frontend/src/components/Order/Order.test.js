import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import Order from './Order';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj: {
        id: 5,
        user_id: 1,
        createdAt: '2019-04-22T10:29:35.000Z',
        updatedAt: '2019-04-22T10:29:35.000Z',
        contents: '[{"idgoods":1,"name":"Honda civic","description":\
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Blanditiis \
        provident quo quaerat repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? \
        Nisi, maxime a.","catalogue_id_catalogue":4,"price":2000},{"idgoods":7,"name":"Opel Omega",\
        "description":"Lorem ipsum dolor sit amet consectetur adipisicing elit. Blanditiis provident \
        quo quaerat repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? Nisi, maxime a.",\
        "catalogue_id_catalogue":4,"price":31225}]'
    }
}) {
    const component = shallow(<Order {...props} />);
    return { component, props };
}

describe('Order', () => {
    it('Component should be rendered', () => {
        const { component } = setupComponent();
        expect(Order.propTypes.obj).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
