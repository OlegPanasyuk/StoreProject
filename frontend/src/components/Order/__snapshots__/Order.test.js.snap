// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Component should be rendered 1`] = `
<Card
  as="div"
  body={false}
  className="mb-3"
>
  <CardHeader
    className="d-flex align-items-center justify-content-between"
  >
    <CardText
      className="mb-0"
    >
      Order #
      5
    </CardText>
    <Button
      active={false}
      aria-controls="collapse-cards-5"
      aria-expanded={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      <i
        className="fas fa-chevron-down"
      />
    </Button>
  </CardHeader>
  <CardBody
    className="d-flex flex-wrap justify-content-start  p-0"
  >
    <Collapse
      appear={false}
      dimension="height"
      getDimensionValue={[Function]}
      in={false}
      mountOnEnter={false}
      timeout={300}
      unmountOnExit={false}
    >
      <CardGroup>
        <OrderInnerItem
          key="1-Honda civic"
          obj={
            Object {
              "catalogue_id_catalogue": 4,
              "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Blanditiis         provident quo quaerat repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non?         Nisi, maxime a.",
              "idgoods": 1,
              "name": "Honda civic",
              "price": 2000,
            }
          }
        />
        <OrderInnerItem
          key="7-Opel Omega"
          obj={
            Object {
              "catalogue_id_catalogue": 4,
              "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Blanditiis provident         quo quaerat repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? Nisi, maxime a.",
              "idgoods": 7,
              "name": "Opel Omega",
              "price": 31225,
            }
          }
        />
      </CardGroup>
    </Collapse>
  </CardBody>
</Card>
`;
